src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/DateTimeValueResolver.php

```diff
@@ -43,25 +49,29 @@ public function resolve(Request $request, ArgumentMetadata $argument): array
         }
 
         $value = $request->attributes->get($argument->getName());
+        $class = \DateTimeInterface::class === $argument->getType() ? \DateTimeImmutable::class : $argument->getType();
 
-        if ($value instanceof \DateTimeInterface) {
-            return [$value];
+        if (!$value) {
+            if ($argument->isNullable()) {
+                return [null];
+            }
+            if (!$this->clock) {
+                return [new $class];
```

Comment:
Early return without constructor argument to prevent deprecation notice. 
It should be submitted as a bugfix?
```
Deprecated: DateTimeImmutable::__construct(): Passing null to parameter #1 ($datetime) of type string is deprecated
```

Reply:
yes (or using `?? 'now'`)

Reply:
Bugfix here: https://github.com/symfony/symfony/pull/48110

Reply:
You should also backport the fix about converting DateTimeInterface to the right class